### ISSUE TITLE
BuildQueryString is not allowing parameters with a value of false

### ISSUE THREAD

**Issue by jtopjian** [gophercloud/gophercloud#168]
This is due to the `isZero` function doing a [recursive check](https://github.com/gophercloud/gophercloud/blob/master/params.go#L235) when the field is a pointer.

Is this expected behavior? If so, how can something like `projects?enabled=false` be accomplished?

**Comment by jrperritt** [comment]
`*bool`

**Comment by jrperritt** [comment]
~~*bool~~

Ah, hmm. Let me think about it.

**Comment by jrperritt** [comment]
I suspect `bool` is a rare type for a query string value. Maybe we just handle those individually in the `ToListQuery` function.

**Comment by jrperritt** [comment]
or maybe `isZero` is changed to use the value of the pointer.

**Comment by jtopjian** [comment]
```shell
$ grep -R \*bool ~/go/src/github.com/gophercloud/gophercloud | grep q:
```

It looks like there's a good amount being used, but definitely less than other types.

Whichever solution you think is best. 😄

**Comment by jrperritt** [comment]
I think `isZero` is wrong here. For any pointer, `nil` should return `true`, anything else `false`.

**Comment by jtopjian** [comment]
So just change:

```go
return isZero(v.Elem())
```

to

```go
return false
```

?

If so, I can make a PR for this as well as confirm I see the expected actions in my Keystone Project branches.

**Comment by jrperritt** [comment]
Yes (and appropriate additions to the already-existing unit test for `isZero`). I suspect I wrote it recursively for some reason, but I can't remember why (let this be a documentation lesson for me). If any tests break because of the change, we can consider changing the types causing the issues.

**Comment by jrperritt** [comment]
closed by #181
